cmake/SetupBlaze.cmake

```diff
@@ -16,3 +16,14 @@ set_property(TARGET Blaze PROPERTY
   INTERFACE_INCLUDE_DIRECTORIES ${BLAZE_INCLUDE_DIR})
 set_property(TARGET Blaze PROPERTY
   INTERFACE_LINK_LIBRARIES Lapack)
+
+add_interface_lib_headers(
+  TARGET Blaze
+  HEADERS
+  blaze/math/CustomVector.h
+  blaze/math/DynamicMatrix.h
+  blaze/math/DynamicVector.h
+  blaze/system/Optimizations.h
+  blaze/system/Version.h
+  blaze/util/typetraits/RemoveConst.h
```

Comment:
Do I understand correctly that tracking the headers of imported targets is needed only for the dependency checking?
Then perhaps it would be less of a maintenance burden to specify one or multiple regexes such as `blaze/*.h` or `pup*.h` instead of each header individually?

Reply:
I'm not generally comfortable on relying on wildcards. `blaze/*.h` might be fine, but `pup*.h` certainly might not be. Also, what if Charm++ instead used `Pup.hpp` and `PupStl.hpp`, then it would conflict with our own `PupStlCpp11.hpp`. While this isn't the case right now, it's going to be really annoying if such a case comes up and someone (most likely me) has to figure out what is happening. That's way more work than adding a header file here every once in a while. Furthermore, CMake strongly advises against wildcards for matching files.

Reply:
I was thinking a list of strings to specify the headers as you have it implemented now, but they are interpreted as regexes. So where you feel safe to use a wildcard or a regex pattern, e.g. for `blaze/*.h` you can do so, but where you don't you just don't use any wildcards in the string, e.g. `pup.h pup_stl.h`. What do you think?

Reply:
No, I'm not comfortable enabling that.

Reply:
To expand a bit, you still have the same potential issue. If we have a library that we stick in a dir `src/blaze` we would end up with the same horrible to debug issue. Again, it's unlikely, but if this happens in 6 years from now it's going to be really annoying to deal with because the current CMake dependency checking code's internals will have been completely forgotten by then.